vyper/semantics/validation/annotation.py

```diff
@@ -55,12 +55,7 @@ def visit(self, node):
         super().visit(node)
 
     def visit_Attribute(self, node):
-        if node.get("value.id") == "msg" and node.attr == "data":
-            parent = node.get_ancestor()
-            if parent.get("func.id") == "slice":
-                node._metadata["size"] = parent.args[1].value + parent.args[2].value
-            elif parent.get("func.id") == "len":
-                node._metadata["is_len"] = True
+        pass
```

Comment:
I think if you just remove this, by default it should work

Reply:
Just tried it out. It does work if I also change size annotation to be `start+32` in the case of `extract32`. But also it fails if I provide a negative start index :/ which sucks since I'd like to be able to grab stuff from the end, without calculating each time the expected size of calldata (or in the case I have a default array type argument I'm really boned, ex. `def foo(..., _data: Bytes[2048] = b"")`)